libcudacxx/include/cuda/std/detail/libcxx/include/version

```diff
@@ -219,6 +219,16 @@ __cpp_lib_void_t                                        201411L <type_traits>
 #pragma GCC system_header
 #endif
 
+// We define certain feature test macros depending on availability. When
+// __CUDA_MINIMUM_ARCH__ is not available, we define the following features
+// unconditionally. This permits checking for the feature in host code. When
+// __CUDA_MINIMUM_ARCH__ is available, we only enable the feature when the
+// hardware supports it.
+#if (!defined(__CUDA_MINIMUM_ARCH__)) || (defined(__CUDA_MINIMUM_ARCH__) && 900 <= __CUDA_MINIMUM_ARCH__)
```

Comment:
question: @griwes @wmaxey Do we want to move such conditions into `__config` and give it a proper name like `_LIBCUDACXX_REQUIRES_SM_90`